distributed/scheduler.py

```diff
@@ -2340,8 +2340,9 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
         if ts._dependencies or valid_workers is not None:
             ws = decide_worker(
                 ts,
-                self._workers_dv.values(),
+                set(self._workers_dv.values()),
```

Comment:
AIUI this could be kind of expensive if we don't need it, which is why the original code didn't do this. However it was that way even prior to Cythonizing the Scheduler